CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.18.1 FATAL_ERROR)
```

Comment:
Maybe we should set minimum version to 3.7 to support some old systems, would we require any modern CMake features?

Ref: https://cliutils.gitlab.io/modern-cmake/chapters/intro/dodonot.html

Reply:
3.7 seems to be a good choice for minimum. Let me change that.